processor/src/chiplets/hasher/tests.rs

```diff
@@ -22,7 +22,8 @@ fn hasher_permute() {
     // initialize the hasher and perform one permutation
     let mut hasher = Hasher::default();
     let init_state: HasherState = rand_array();
-    let (addr, final_state, _) = hasher.permute(init_state);
+    let mut lookups = Vec::new();
+    let (addr, final_state) = hasher.permute(init_state, &mut lookups);
```

Comment:
It would probably be good to validate the lookups in these unit tests. It doesn't need to be done as part of this PR, but I think it might be helpful to have before continuing to the next stage of the hasher refactor where things get more complicated

Reply:
I've added the validity checks for lookups in the tests. Please let me know if I should move the tests to another PR if it's better to merge this one without the new changes first.

Reply:
Adding them in this one is fine